diff -Nurp a/po/Makefile b/po/Makefile
--- a/po/Makefile	2015-08-05 08:42:21.000000000 +0200
+++ b/po/Makefile	2016-05-08 21:33:15.939665392 +0200
@@ -50,7 +50,7 @@ generated += $(ALL_LINGUAS:%=%.mo)
 include ../Rules
 
 # Generate the translation template from all the source files.
-libc.pot: pot.header libc.pot.files
+libc.pot: pot.header $(objpfx)libc.pot.files
 	@rm -f $@.new
 	set `date -R`; disp="$$6"; \
 	sed -e 's/VERSION/$(version)/' \
@@ -74,7 +74,7 @@ po-sed-cmd = \
   '/\/tst-/d;$(foreach S,[ch] cxx sh ksh bash,$(subst .,\.,/.$S\(.in\)*$$/p;))'
 
 # Omit tst-* files, and get only files with the given suffixes.
-libc.pot.files: FORCE
+$(objpfx)libc.pot.files: FORCE
 	$(..)scripts/list-sources.sh .. | sed -n $(po-sed-cmd) > $@.new
 	mv -f $@.new $@
 
