Index: autoconf/bin/autoconf.as
===================================================================
--- autoconf.orig/bin/autoconf.as	2024-01-01 11:21:50.522251005 +0100
+++ autoconf/bin/autoconf.as	2024-01-01 14:06:26.729053991 +0100
@@ -89,8 +89,8 @@
 # restore font-lock: '
 
 # Variables.
-: ${AUTOM4TE='@bindir@/@autom4te-name@'}
-: ${trailer_m4='@pkgdatadir@/autoconf/trailer.m4'}
+: ${AUTOM4TE='@autom4te-name@'}
+: ${trailer_m4="$(realpath -m "$(dirname "$as_myself")/../share/autoconf/autoconf/trailer.m4")"}
 autom4te_options=
 outfile=
 verbose=false
Index: autoconf/bin/autoheader.in
===================================================================
--- autoconf.orig/bin/autoheader.in	2024-01-01 11:21:50.522251005 +0100
+++ autoconf/bin/autoheader.in	2024-01-01 11:21:50.518250939 +0100
@@ -32,7 +32,10 @@
 
 BEGIN
 {
-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
+  use File::Basename;
+  use File::Spec;
+  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf');
   unshift @INC, "$pkgdatadir";
 
   # Override SHELL.  On DJGPP SHELL may not be set to a shell
@@ -54,7 +57,8 @@
 our ($config_h, %symbol, %verbatim);
 
 # Lib files.
-my $autom4te = $ENV{'AUTOM4TE'} || '@bindir@/@autom4te-name@';
+my $prefix = File::Basename::dirname(File::Spec->rel2abs($0)) . '/..';
+my $autom4te = $ENV{'AUTOM4TE'} || "$prefix/bin/@autom4te-name@";
 my $config_h_in;
 my @prepend_include;
 my @include;
Index: autoconf/bin/autom4te.in
===================================================================
--- autoconf.orig/bin/autom4te.in	2024-01-01 11:21:50.522251005 +0100
+++ autoconf/bin/autom4te.in	2024-01-01 11:21:50.518250939 +0100
@@ -28,7 +28,10 @@
 
 BEGIN
 {
-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
+  use File::Basename;
+  use File::Spec;
+  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf');
   unshift @INC, $pkgdatadir;
 
   # Override SHELL.  On DJGPP SHELL may not be set to a shell
@@ -48,7 +51,8 @@
 use Autom4te::XFile;
 
 # Data directory.
-my $pkgdatadir = $ENV{'AC_MACRODIR'} || '@pkgdatadir@';
+my $prefix = File::Basename::dirname(File::Spec->rel2abs($0)) . '/..';
+my $pkgdatadir = $ENV{'AC_MACRODIR'} || "$prefix/share/autoconf";
 
 # $LANGUAGE{LANGUAGE} -- Automatic options for LANGUAGE.
 my %language;
@@ -288,7 +292,11 @@
 	{
 	  fatal "$file:$.: no current language"
 	    unless defined $lang;
-	  push @{$language{$lang}}, @words;
+	  foreach (@words)
+	    {
+	      s/~/$prefix/; # make relocatable
+	      push @{$language{$lang}}, $_;
+	    }
 	}
       else
 	{
Index: autoconf/bin/autoreconf.in
===================================================================
--- autoconf.orig/bin/autoreconf.in	2024-01-01 11:21:50.522251005 +0100
+++ autoconf/bin/autoreconf.in	2024-01-01 11:21:50.522251005 +0100
@@ -32,10 +32,14 @@
 my $buildauxdir;
 BEGIN
 {
-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
+  use File::Basename;
+  use File::Spec;
+  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf');
   unshift @INC, $pkgdatadir;
 
-  $buildauxdir = $ENV{'autom4te_buildauxdir'} || $pkgdatadir . '/build-aux';
+  $buildauxdir = $ENV{'autom4te_buildauxdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf', 'build-aux');
 
   # Override SHELL.  On DJGPP SHELL may not be set to a shell
   # that can handle redirection and quote arguments correctly,
@@ -117,9 +121,10 @@
 ";
 
 # Lib files.
-my $autoconf    = $ENV{'AUTOCONF'}    || '@bindir@/@autoconf-name@';
-my $autoheader  = $ENV{'AUTOHEADER'}  || '@bindir@/@autoheader-name@';
-my $autom4te    = $ENV{'AUTOM4TE'}    || '@bindir@/@autom4te-name@';
+my $prefix = File::Basename::dirname(File::Spec->rel2abs($0)) . '/..';
+my $autoconf    = $ENV{'AUTOCONF'}    || "$prefix/bin/@autoconf-name@";
+my $autoheader  = $ENV{'AUTOHEADER'}  || "$prefix/bin/@autoheader-name@";
+my $autom4te    = $ENV{'AUTOM4TE'}    || "$prefix/bin/@autom4te-name@";
 my $automake    = $ENV{'AUTOMAKE'}    || 'automake';
 my $aclocal     = $ENV{'ACLOCAL'}     || 'aclocal';
 my $libtoolize  = $ENV{'LIBTOOLIZE'}  || 'libtoolize';
Index: autoconf/bin/autoscan.in
===================================================================
--- autoconf.orig/bin/autoscan.in	2024-01-01 11:21:50.522251005 +0100
+++ autoconf/bin/autoscan.in	2024-01-01 11:21:50.522251005 +0100
@@ -30,7 +30,10 @@
 
 BEGIN
 {
-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
+  use File::Basename;
+  use File::Spec;
+  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf');
   unshift @INC, $pkgdatadir;
 
   # Override SHELL.  On DJGPP SHELL may not be set to a shell
@@ -95,10 +98,11 @@
 my $log;
 
 # Autoconf and lib files.
-my $autom4te = $ENV{'AUTOM4TE'} || '@bindir@/@autom4te-name@';
+my $prefix = File::Basename::dirname(File::Spec->rel2abs($0)) . '/..';
+my $autom4te = $ENV{'AUTOM4TE'} || "$prefix/bin/@autom4te-name@";
 my $autoconf = "$autom4te --language=autoconf";
 my @prepend_include;
-my @include = ('@pkgdatadir@');
+my @include = ("$prefix/share/autoconf");
 
 # $help
 # -----
Index: autoconf/bin/autoupdate.in
===================================================================
--- autoconf.orig/bin/autoupdate.in	2024-01-01 11:21:50.522251005 +0100
+++ autoconf/bin/autoupdate.in	2024-01-01 11:21:50.522251005 +0100
@@ -31,7 +31,10 @@
 
 BEGIN
 {
-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
+  use File::Basename;
+  use File::Spec;
+  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf');
   unshift @INC, $pkgdatadir;
 
   # Override SHELL.  On DJGPP SHELL may not be set to a shell
@@ -51,14 +54,15 @@
 use Autom4te::XFile;
 
 # Lib files.
-my $autom4te = $ENV{'AUTOM4TE'} || '@bindir@/@autom4te-name@';
+my $prefix = File::Basename::dirname(File::Spec->rel2abs($0)) . '/..';
+my $autom4te = $ENV{'AUTOM4TE'} || "$prefix/bin/@autom4te-name@";
 my $autoconf = "$autom4te --language=autoconf";
 # We need to find m4sugar.
 my @prepend_include;
-my @include = ('@pkgdatadir@');
+my @include = ("$prefix/share/autoconf");
 my $force = 0;
 # m4.
-my $m4 = $ENV{"M4"} || '@M4@';
+my $m4 = $ENV{"M4"} || 'm4';
 
 
 # $HELP
Index: autoconf/bin/ifnames.in
===================================================================
--- autoconf.orig/bin/ifnames.in	2024-01-01 11:21:50.522251005 +0100
+++ autoconf/bin/ifnames.in	2024-01-01 11:21:50.522251005 +0100
@@ -35,7 +35,10 @@
 
 BEGIN
 {
-  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || '@pkgdatadir@';
+  use File::Basename;
+  use File::Spec;
+  my $pkgdatadir = $ENV{'autom4te_perllibdir'} || File::Spec->catdir(
+    File::Basename::dirname(File::Spec->rel2abs($0)), '..', 'share', 'autoconf');
   unshift @INC, $pkgdatadir;
 
   # Override SHELL.  On DJGPP SHELL may not be set to a shell
Index: autoconf/lib/autom4te.in
===================================================================
--- autoconf.orig/lib/autom4te.in	2024-01-01 11:21:50.522251005 +0100
+++ autoconf/lib/autom4te.in	2024-01-01 11:21:50.522251005 +0100
@@ -115,7 +115,7 @@
 # This intermediate language is used by aclocal to build aclocal.m4.
 
 begin-language: "Autoconf-without-aclocal-m4"
-args: --prepend-include '@pkgdatadir@'
+args: --prepend-include '~/share/autoconf'
 args: --cache=autom4te.cache
 args: autoconf/autoconf.m4f
 args: acsite.m4?
@@ -142,7 +142,7 @@
 ## -------- ##
 
 begin-language: "Autotest"
-args: --prepend-include '@pkgdatadir@'
+args: --prepend-include '~/share/autoconf'
 args: autotest/autotest.m4f
 args: package.m4?
 args: local.at?
@@ -156,7 +156,7 @@
 ## ---- ##
 
 begin-language: "M4sh"
-args: --prepend-include '@pkgdatadir@'
+args: --prepend-include '~/share/autoconf'
 args: m4sugar/m4sh.m4f
 args: --mode 777
 args: --language M4sugar
@@ -168,6 +168,6 @@
 ## ------- ##
 
 begin-language: "M4sugar"
-args: --prepend-include '@pkgdatadir@'
+args: --prepend-include '~/share/autoconf'
 args: m4sugar/m4sugar.m4f
 end-language: "M4sugar"
