diff -ur a/lib/getlocalename_l-unsafe.c b/lib/getlocalename_l-unsafe.c
--- a/lib/getlocalename_l-unsafe.c	2025-05-10 14:41:50.000000000 +0200
+++ b/lib/getlocalename_l-unsafe.c	2025-10-07 20:39:27.485874173 +0200
@@ -34,7 +34,7 @@
 #endif
 #include "setlocale_null.h"
 
-#if (__GLIBC__ >= 2 && !defined __UCLIBC__) || (defined __linux__ && HAVE_LANGINFO_H) || defined __CYGWIN__
+#if (__GLIBC__ >= 2 && !defined __UCLIBC__) || ((defined(__linux__) || defined(__l4re__)) && HAVE_LANGINFO_H) || defined __CYGWIN__
 # include <langinfo.h>
 #endif
 #if defined __sun
@@ -484,7 +484,7 @@
            nl_langinfo_l (_NL_LOCALE_NAME (category), locale).  */
         name = locale->__names[category];
       return (struct string_with_storage) { name, STORAGE_OBJECT };
-#elif defined __linux__ && HAVE_LANGINFO_H && defined NL_LOCALE_NAME
+#elif (defined(__linux__) || defined(__l4re__)) && HAVE_LANGINFO_H && defined NL_LOCALE_NAME
       /* musl libc */
       const char *name = nl_langinfo_l (NL_LOCALE_NAME (category), locale);
       return (struct string_with_storage) { name, STORAGE_OBJECT };
