--- meson_options.txt.orig	2020-12-11 14:04:57.478605095 +0100
+++ meson_options.txt	2020-12-11 14:05:17.222892037 +0100
@@ -115,4 +115,9 @@ option('glib_checks',
 option('libelf',
        type : 'feature',
        value : 'auto',
-       description : 'Enable support for listing and extracting from ELF resource files with gresource tool')
\ Kein Zeilenumbruch am Dateiende.
+       description : 'Enable support for listing and extracting from ELF resource files with gresource tool')
+
+option('tests',
+       type : 'boolean',
+       value : false,
+       description : 'Build tests')
--- meson.build.orig	2020-12-11 14:05:29.207066017 +0100
+++ meson.build	2020-12-11 14:05:38.127195440 +0100
@@ -93,8 +93,7 @@ installed_tests_enabled = get_option('in
 installed_tests_template = files('template.test.in')
 installed_tests_template_tap = files('template-tap.test.in')
 
-# Don’t build the tests unless we can run them (either natively, in an exe wrapper, or by installing them for later use)
-build_tests = not meson.is_cross_build() or (meson.is_cross_build() and meson.has_exe_wrapper()) or installed_tests_enabled
+build_tests = get_option('tests')
 
 add_project_arguments('-D_GNU_SOURCE', language: 'c')
 
