diff -urN a/configure b/configure
--- a/configure	2025-08-15 12:40:25.114956329 +0200
+++ b/configure	2025-08-15 12:41:28.599194193 +0200
@@ -3345,6 +3345,7 @@
 *-genode*)	opt_bash_malloc=no ;;	# Genode has no sbrk
 i370-openedition*)	opt_bash_malloc=no LOCAL_LIBS=-lzoslib LIBS_FOR_BUILD=-lzoslib ;;	# IBM z/OS machines
 i370-*)		opt_bash_malloc=no ;;	# generic (?) IBM 370 machines
+*-l4re*)	opt_bash_malloc=no ;;	# L4Re's malloc is great
 esac
 
 # memory scrambling on free()
